File: evidence/generated-systems/golang/hr-on-golang/hr-on-golang/internal/model/Position.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Position Declaration
//==============================================================
type Position struct {
    gorm.Model
     PositionCode                                    string
    Fte                                                            string
    DepartmentId         *uint
    Department           *Department `gorm:"foreignKey:DepartmentId"`
    JobProfileId         *uint
    JobProfile           *JobProfile `gorm:"foreignKey:JobProfileId"`
    CostCenterId         *uint
    CostCenter           *CostCenter `gorm:"foreignKey:CostCenterId"`
    LocationId         *uint
    Location           *Location `gorm:"foreignKey:LocationId"`
    ManagerPositionId         *uint
    ManagerPosition           *Position `gorm:"foreignKey:ManagerPositionId"`
     DirectReports           []Position `gorm:"foreignKey:DirectReportsFromPositionId"`
     Assignments           []EmploymentAssignment `gorm:"foreignKey:AssignmentsFromPositionId"`
    Status                      PositionStatus
    WorkLocationType                      WorkLocationType

// parent associations as their child

}

File: evidence/generated-systems/angular/hr-on-angular/src/app/models/Position.js

// Define collection and schema for Position
export interface Position {
    positionCode:
	type : string
    fte:
	type : String
    Department:
	type : Schema.Types.ObjectId
    JobProfile:
	type : Schema.Types.ObjectId
    CostCenter:
	type : Schema.Types.ObjectId
    Location:
	type : Schema.Types.ObjectId
    ManagerPosition:
	type : Schema.Types.ObjectId
    DirectReports:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Position' }]
    Assignments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'EmploymentAssignment' }]
    Status:
 	type : String
    WorkLocationType:
 	type : String
#
    collection: 'positions'
}
